control-plane/connect-inject/peering_dialer_controller_test.go

```diff
@@ -18,6 +23,462 @@ import (
+							Secret: v1alpha1.Secret{
+								Name:    "dialer-token-old",
+								Key:     "token",
+								Backend: "kubernetes",
+							},
+							ResourceVersion: "dummy-version",
+						},
+					},
+				}
+				return []runtime.Object{dialer}
+			},
+			expectedConsulPeerings: &api.Peering{
+				Name:  "peering",
+				State: api.PeeringStateActive,
+			},
+			peeringSecret: func(token string) *corev1.Secret {
+				return createSecret("dialer-token", "default", "token", token)
+			},
+			expectedStatus: &v1alpha1.PeeringDialerStatus{
+				SecretRef: &v1alpha1.SecretRefStatus{
+					Secret: v1alpha1.Secret{
+						Name:    "dialer-token",
+						Key:     "token",
+						Backend: "kubernetes",
+					},
+				},
+			},
+			peeringExists: true,
+		},
+	}
+	for name, tt := range cases {
+		t.Run(name, func(t *testing.T) {
+
+			// Create test consul server
+			acceptorPeer, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+				c.NodeName = nodeName
+			})
+			require.NoError(t, err)
+			defer acceptorPeer.Stop()
+			acceptorPeer.WaitForServiceIntentions(t)
+
+			cfg := &api.Config{
+				Address: acceptorPeer.HTTPAddr,
+			}
+			acceptorClient, err := api.NewClient(cfg)
+			require.NoError(t, err)
+
+			// Add the default namespace.
+			ns := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}
+			// Create fake k8s client
+			k8sObjects := append(tt.k8sObjects(), &ns)
+
+			// This is responsible for updating the token generated by the acceptor side with the IP
+			// of the Consul server as the generated token currently does not have that set on it.
+			var encodedPeeringToken string
+			if tt.peeringName != "" {
+				var token struct {
+					CA              string
+					ServerAddresses []string
+					ServerName      string
+					PeerID          string
+				}
+				// Create the initial token,
+				baseToken, _, err := acceptorClient.Peerings().GenerateToken(context.Background(), api.PeeringGenerateTokenRequest{PeerName: tt.peeringName}, nil)
+				require.NoError(t, err)
+				// Decode the token to extract the ServerName and PeerID from the token. CA is always NULL.
+				decodeBytes, err := base64.StdEncoding.DecodeString(baseToken.PeeringToken)
+				require.NoError(t, err)
+				err = json.Unmarshal(decodeBytes, &token)
+				require.NoError(t, err)
+				// Get the IP of the Consul server.
+				addr := strings.Split(acceptorPeer.HTTPAddr, ":")[0]
+				// Generate expected token for Peering Initiate.
+				tokenString := fmt.Sprintf(`{"CA":null,"ServerAddresses":["%s:8300"],"ServerName":"%s","PeerID":"%s"}`, addr, token.ServerName, token.PeerID)
+				// Create peering initiate secret in Kubernetes.
+				encodedPeeringToken = base64.StdEncoding.EncodeToString([]byte(tokenString))
+				k8sObjects = append(k8sObjects, tt.peeringSecret(encodedPeeringToken))
+			}
+
+			// Create test consul server
+			diallerPeer, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
```

Comment:
`diallerPeer` is misspelled in this function, should be `dialerPeer` ?

Reply:
same with client. honestly its so nit maybe not worth bothering :D

Reply:
If you do go back and change it. Can we call it `dialerPeerServer` or something?

Reply:
+1 to `dialerPeerServer` :)